sniffy-core/src/main/java/io/sniffy/socket/SnifferOutputStream.java

```diff
@@ -81,7 +85,22 @@ private void sleepIfRequired(int bytesUp) throws ConnectException {
 
     @Override
     public void write(byte[] b) throws IOException {
+
         if (null != snifferSocket) snifferSocket.checkConnectionAllowed(0);
+
+        if (null != snifferSocket && !snifferSocket.isFirstPacketSent()) {
+
+            try {
+                SniffyPacketAnalyzer sniffyPacketAnalyzer = new SniffyPacketAnalyzer(snifferSocket);
```

Comment:
Wrap proxy detection functionality with a feature flag / config switch and document it